### PR TITLE
bugfix: dubbo reference  annotation failed to register transaction on tcc mode

### PR DESCRIPTION
**Pull request by zhongfuhua**
### Ⅰ. Describe what this PR did
Unable to register transaction branch in TCC transaction using Dubbo annotation @reference

### Ⅳ. Describe how to verify it

Please test the springboot-tcc-sample module of seata-sample

### CODE REVIEW

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java [hunk 1]
```diff
@@ -63,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         } finally {
             if (bind) {
                 String unbindXid = RootContext.unbind();
+                String unbindFilterType = RootContext.unbindFilterType();
                 if (LOGGER.isDebugEnabled()) {
-                    LOGGER.debug("unbind[" + unbindXid + "] from RootContext");
+                    LOGGER.debug("unbind[{}] from RootContext", unbindXid);
+                    LOGGER.debug("unbind filterType[{}] from RootContext", unbindFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java [hunk 2]
```diff
@@ -41,19 +40,25 @@
...
+            RpcContext.getContext().setAttachment(RootContext.KEY_XID_FILTER_TYPE, xidFilterType);
         } else {
             if (rpcXid != null) {
                 RootContext.bind(rpcXid);
+                RootContext.bindFilterType(rpcXidFilterType);
                 bind = true;
                 if (LOGGER.isDebugEnabled()) {
-                    LOGGER.debug("bind[" + rpcXid + "] to RootContext");
+                    LOGGER.debug("bind[{}] to RootContext", rpcXid);
+                    LOGGER.debug("bind filterType[{}] to RootContext", rpcXidFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: merge logs.

Comment (xingfudeshi) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java [hunk 3]
```diff
@@ -63,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
...
+                String unbindFilterType = RootContext.unbindFilterType();
                 if (LOGGER.isDebugEnabled()) {
-                    LOGGER.debug("unbind[" + unbindXid + "] from RootContext");
+                    LOGGER.debug("unbind[{}] from RootContext", unbindXid);
+                    LOGGER.debug("unbind filterType[{}] from RootContext", unbindFilterType);
                 }
                 if (!rpcXid.equalsIgnoreCase(unbindXid)) {
-                    LOGGER.warn("xid in change during RPC from " + rpcXid + " to " + unbindXid);
+                    LOGGER.warn("xid in change during RPC from {} to {}", rpcXid, unbindXid);
+                    LOGGER.warn("xidFilterType in change during RPC from {} to {}", rpcXidFilterType, unbindFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java [hunk 4]
```diff
@@ -63,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
...
                     if (unbindXid != null) {
                         RootContext.bind(unbindXid);
-                        LOGGER.warn("bind [" + unbindXid + "] back to RootContext");
+                        RootContext.bindFilterType(unbindFilterType);
+                        LOGGER.warn("bind [{}] back to RootContext", unbindXid);
+                        LOGGER.warn("bind filterType [{}] back to RootContext", unbindFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 5]
```diff
@@ -40,19 +40,25 @@
...
+            RpcContext.getContext().setAttachment(RootContext.KEY_XID_FILTER_TYPE, xidFilterType);
         } else {
             if (rpcXid != null) {
                 RootContext.bind(rpcXid);
+                RootContext.bindFilterType(rpcXidFilterType);
                 bind = true;
                 if (LOGGER.isDebugEnabled()) {
-                    LOGGER.debug("bind[" + rpcXid + "] to RootContext");
+                    LOGGER.debug("bind[{}] to RootContext", rpcXid);
+                    LOGGER.debug("bind filterType[{}] to RootContext", rpcXidFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 6]
```diff
@@ -62,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
         } finally {
             if (bind) {
                 String unbindXid = RootContext.unbind();
+                String unbindFilterType = RootContext.unbindFilterType();
                 if (LOGGER.isDebugEnabled()) {
-                    LOGGER.debug("unbind[" + unbindXid + "] from RootContext");
+                    LOGGER.debug("unbind[{}] from RootContext", unbindXid);
+                    LOGGER.debug("unbind filterType[{}] from RootContext", unbindFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 7]
```diff
@@ -62,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
...
                 }
                 if (!rpcXid.equalsIgnoreCase(unbindXid)) {
-                    LOGGER.warn("xid in change during RPC from " + rpcXid + " to " + unbindXid);
+                    LOGGER.warn("xid in change during RPC from {} to {}", rpcXid, unbindXid);
+                    LOGGER.warn("xidFilterType in change during RPC from {} to {}", rpcXidFilterType, unbindFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java [hunk 8]
```diff
@@ -62,14 +68,19 @@ public Result invoke(Invoker<?> invoker, Invocation invocation) throws RpcExcept
...
                     if (unbindXid != null) {
                         RootContext.bind(unbindXid);
-                        LOGGER.warn("bind [" + unbindXid + "] back to RootContext");
+                        RootContext.bindFilterType(unbindFilterType);
+                        LOGGER.warn("bind [{}] back to RootContext", unbindXid);
+                        LOGGER.warn("bind filterType [{}] back to RootContext", unbindFilterType);
```
Comment (xingfudeshi) [inline integration/dubbo/src/main/java/io/seata/integration/dubbo/TransactionPropagationFilter.java]: merge logs

File: integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java (hunk 1 above)
Comment (zjinlei) [reply integration/dubbo-alibaba/src/main/java/io/seata/integration/dubbo/alibaba/TransactionPropagationFilter.java]: @xingfudeshi pr1813 include these codes and logs merged,can reveiw 1813 first.

**Comment by zjinlei** [comment]
please resolve conflicts

File: core/src/main/java/io/seata/core/context/RootContext.java [hunk 9]
```diff
@@ -39,6 +39,8 @@
 
     public static final String KEY_XID_INTERCEPTOR_TYPE = "tx-xid-interceptor-type";
 
+    public static final String KEY_XID_FILTER_TYPE = "tx-xid-filter-type";
```
Comment (slievrly) [inline core/src/main/java/io/seata/core/context/RootContext.java]: why need this?

**Comment by zhangthen** [comment]
这个只支持了dubbo，不支持其他rpc框架，不是一个通用方案，我安排人 开发一个 通用方案吧

**Comment by funky-eyes** [comment]
https://github.com/seata/seata/pull/5399

**Pull request closed without merging**